sql/catalyst/src/main/scala/org/apache/spark/sql/types/DecimalType.scala

```diff
@@ -76,23 +77,29 @@ case class DecimalType(precision: Int, scale: Int) extends FractionalType {
    * Returns whether this DecimalType is wider than `other`. If yes, it means `other`
    * can be casted into `this` safely without losing any precision or range.
    */
-  private[sql] def isWiderThan(other: DataType): Boolean = other match {
+  private[sql] def isWiderThan(other: DataType): Boolean = isWiderThanInternal(other)
```

Comment:
Otherwise:

```
[error] /.../spark/sql/catalyst/src/main/scala/org/apache/spark/sql/types/DecimalType.scala:81:20: could not optimize @tailrec annotated method isWiderThan: it is neither private nor final so can be overridden
[error]   private[sql] def isWiderThan(other: DataType): Boolean = other match {
[error]                    ^
```